client/my-sites/domains/components/domain-and-plan-package/navigation/index.jsx

```diff
@@ -9,7 +8,7 @@ export default function DomainAndPlanPackageNavigation( props ) {
 
 	const goBack = () => {
 		if ( props.goBackLink ) {
-			page( props.goBackLink );
```

Comment:
It looks like window.location.assign works here. But I do think page() and window.location.assign() are directly equal. The page() method seems to be an abstraction with some extra routing logic. I think it could theoretically accept arguments/values that would not work with window.location.assign(). I did a quick search to see what kinds of other values are passed as props.goBackLink. From what I can tell, they seem like they should all still work fine here. But still wanted to note this.

Reply:
Another thing I can do is I can pass down a `isLaunchpad` prop and conditionally decide whether to use `page` or `location.assign` based on it

Reply:
As a side note - using TypeScript in this file and destructing 'props' might help provide clarity on what values and types are expected for goBackLink. Obviously not related to this specific PR.